instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import Comment from "../Comment/Comment";
+import "../CommentSection/CommentSection.css";
+import moment from "moment";
+
+class CommentSection extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.state = {
+      comment: "",
+      likes: props.post.likes,
+      id: props.post.id,
+      comments: [
+        ...props.comments,
+        ...props.commentEntries.filter(entry => entry.id === props.post.id),
+        
+      ]};
+  }
+
+  
+
+  addComment = event => {
+    event.preventDefault();
+    this.setState(
+      {
+        comments: [
+          ...this.state.comments,
+          { id: this.state.id, text: this.state.comment, username: "dummyUser" }
+        ],
+        comment: ""
+      },
+      () =>
+        this.props.updateComments(
+          this.state.comments[this.state.comments.length - 1]
+        )
+    );
+  };
+
+  commentInput = event => {
+    this.setState({
+      comment: event.target.value
+    });
+  };
+
+  addLike = () => {
+    let likeCounter = this.state.likes;
+    this.setState({
+      likes: ++likeCounter
+    });
+  };
+
+  render() {
+    let date = moment(
+      this.props.post.timestamp,
+      "MMMM Do YYYY, hh:mm:ss a"
+    ).format("YYYY-MM-DD HH:mm:ss");
+    
+    // console.log(entries)
+    return (
+      <div className={"commentsContainer"}>
+        <div className={"buttonBar"}>
+          <div className={"commentsButtonBox"}>
+            <img
+              src="https://png.icons8.com/ios/48/000000/hearts.png"
+              alt=""
+              onClick={this.addLike}
+            />
+            <img
+              src="https://png.icons8.com/ios/48/000000/speech-bubble.png"
+              className={"commentBubble"}
+              alt=""
+            />
+            <img src="https://png.icons8.com/ios/48/000000/upload.png" alt="" />
+          </div>
+          <img
+            src="https://png.icons8.com/ios/48/000000/bookmark-ribbon.png"
+            alt=""
+          />
+        </div>
+        <div className={"likesCount"}>{this.state.likes} likes</div>
+        {this.state.comments.map(comment => {
+          return (
+            <Comment
+              key={Math.random()}
```

Comment:
You could access each comments index from the comments array instead of using Math.random to generate a key. Another idea would be to slice off n amount of letters from a comment as a key, but here's some more info about Keys straight from Reacts documentation:

`The best way to pick a key is to use a string that uniquely identifies a list item among its siblings. Most often you would use IDs from your data as keys:`
```
const todoItems = todos.map((todo) =>
  <li key={todo.id}>
    {todo.text}
  </li>
);

```
`When you don’t have stable IDs for rendered items, you may use the item index as a key as a last resort:`